lib/ipv4/ipv4_packet.ml

```diff
@@ -94,23 +97,24 @@ module Marshal = struct
     set_ipv4_proto buf t.proto;
     set_ipv4_src buf (Ipaddr.V4.to_int32 t.src);
     set_ipv4_dst buf (Ipaddr.V4.to_int32 t.dst);
-    Cstruct.blit t.options 0 buf sizeof_ipv4 (Cstruct.len t.options)
+    Cstruct.blit t.options 0 buf sizeof_ipv4 (Cstruct.len t.options);
+    set_ipv4_len buf (sizeof_ipv4 + (options_len / 4) + (Cstruct.len payload))
```

Comment:
nitpick: the `options / 4` makes me cringe.  the same expression is 6 lines above... maybe instead of `nearest_4` we should have some word-length-options?